Modules/_sqlite/cursor.c

```diff
@@ -750,47 +744,37 @@ pysqlite_cursor_executescript_impl(pysqlite_Cursor *self,
         return NULL;
     }
 
-    /* commit first */
-    result = _PyObject_CallMethodIdNoArgs((PyObject *)self->connection, &PyId_commit);
-    if (!result) {
-        goto error;
-    }
-    Py_DECREF(result);
-
-    pysqlite_state *state = self->connection->state;
-    while (1) {
-        const char *tail;
+    // Commit if needed
+    sqlite3 *db = self->connection->db;
```

Comment:
Hummmm, I am missing something. How we are skipping a *python call* to "commit" on the connection?

Reply:
The `_PyObject_CallMethodIdNoArgs` call on the connection in line 754 is now done using `sqlite3_exec`. We're not skipping the commit; we just execute it using SQLite API iso. going through the `_PyObject_CallMethodIdNoArgs` call.

https://github.com/python/cpython/blob/86d8b465231473f850cc5e906013ba8581ddb503/Modules/_sqlite/cursor.c#L754-L758

Reply:
The added test exercise this branch. We could add that test to the `sqlite3` test suite first, so we have better code coverage for `executescript()` before proceeding with this PR.

Reply:
> We could add that test to the `sqlite3` test suite first, so we have better code coverage for `executescript()` before proceeding with this PR.

👍

Reply:
I'll fix that!

Reply:
Merged!

Reply:
Rebased onto `main` :)